### ISSUE TITLE
Use Classifier cifar

### ISSUE THREAD

**Issue by zaher88abd** [pjreddie/darknet#519]
Hi,

I tried to use the classifier Cifar, but I got this as out put 
`cifar_small
1
layer     filters    size              input                output
    0 conv    128  3 x 3 / 1    52 x  52 x   3   ->    52 x  52 x 128
    1 conv    128  3 x 3 / 1    52 x  52 x 128   ->    52 x  52 x 128
    2 conv    128  3 x 3 / 1    52 x  52 x 128   ->    52 x  52 x 128
    3 max          2 x 2 / 2    52 x  52 x 128   ->    26 x  26 x 128
    4 dropout       p = 0.50               86528  ->  86528
    5 conv    256  3 x 3 / 1    26 x  26 x 128   ->    26 x  26 x 256
    6 conv    256  3 x 3 / 1    26 x  26 x 256   ->    26 x  26 x 256
    7 conv    256  3 x 3 / 1    26 x  26 x 256   ->    26 x  26 x 256
    8 max          2 x 2 / 2    26 x  26 x 256   ->    13 x  13 x 256
    9 dropout       p = 0.50               43264  ->  43264
   10 conv    512  3 x 3 / 1    13 x  13 x 256   ->    13 x  13 x 512
   11 conv    512  3 x 3 / 1    13 x  13 x 512   ->    13 x  13 x 512
   12 conv    512  3 x 3 / 1    13 x  13 x 512   ->    13 x  13 x 512
   13 dropout       p = 0.50               86528  ->  86528
   14 conv     10  1 x 1 / 1    13 x  13 x 512   ->    13 x  13 x  10
   15 avg                       13 x  13 x  10   ->    10
   16 softmax                                          10
   17 type: Using default 'sse'
cost                                             10
Learning Rate: 0.4, Momentum: 0.9, Decay: 0.0005

5411
32 32
Too many or too few labels: 0, data/cifar/train/4729_pretzels.png
Too many or too few labels: 0, data/cifar/train/1090_pretzels.png
Too many or too few labels: 0, data/cifar/train/3496_pretzels.png
Too many or too few labels: 0, data/cifar/train/3069_pretzels.png
Too many or too few labels: 0, data/cifar/train/2970_pretzels.png
Too many or too few labels: 0, data/cifar/train/2842_pretzels.png
Too many or too few labels: 0, data/cifar/train/636_pretzels.png
Too many or too few labels: 0, data/cifar/train/4729_pretzels.png
Too many or too few labels: 0, data/cifar/train/5342_pretzels.png
Too many or too few labels: 0, data/cifar/train/5268_pretzels.png
Too many or too few labels: 0, data/cifar/train/1903_pretzels.png
Too many or too few labels: 0, data/cifar/train/3719_pretzels.png
Too many or too few labels: 0, data/cifar/train/3133_pretzels.png
Too many or too few labels: 0, data/cifar/train/4679_pretzels.png
Too many or too few labels: 0, data/cifar/train/576_pretzels.png
Too many or too few labels: 0, data/cifar/train/128_pretzels.png
Too many or too few labels: 0, data/cifar/train/4624_pretzels.png
Loaded: 0.042833 seconds
`
So, any could help me to know what the meaning of "Too many or too few labels: 0", and just one of the class is used to train even after 500 iterations. And I need to resize the images or the NN will resize them?

**Comment by spaul13** [comment]
@zaher88abd, have u solved the problem of too-many or too few labels?

I am also getting the same problem. Can u plz tell me how to solve this problem?

**Comment by AnhPC03** [comment]
@spaul13 I've investigated the code and had some changes in **fill_truth** *function* in **src/data.c** file. This allow we name root path freedom as we want.
```
void fill_truth(char *path, char **labels, int k, float *truth)
 {
     int i;
     char *filename = "";
     for (i = strlen(path) - 1; i >= 0; --i) {
         if (path[i] == '/' || path[i] == '\\') {
             filename = (char*)malloc(strlen(path + i) + 1);
             strcpy(filename, path + i + 1);
             break;
         }
     }
     memset(truth, 0, k*sizeof(float));
     int count = 0;
     for(i = 0; i < k; ++i){
         if(strstr(filename, labels[i])){
             truth[i] = 1;
             ++count;
             //printf("%s %s %d\n", path, labels[i], i);
         }
     }
     if(count != 1 && (k != 1 || count != 0)) printf("Too many or too few labels: %d, %s\n", count, path);
 }

```
